patroni/postgresql/citus.py

```diff
@@ -293,10 +302,16 @@ def _add_task(self, task: PgDistNode) -> bool:
         with self._condition:
             i = self.find_task_by_group(task.group)
 
-            # task.timeout is None is an indicator that it was scheduled
-            # from the sync_pg_dist_node() and we don't want to override
-            # already existing task created from REST API.
-            if task.timeout is None and (i is not None or self._in_flight and self._in_flight.group == task.group):
+            # The task.timeout is None is an indicator that it was scheduled from the sync_pg_dist_node().
+            # We don't want to override already existing task created from REST API.
+            # In case if there is already _in_flight we don't want to add a task until timeout is reached.
+            # The last one is important to protect from the case when REST API request failed.
+            # But, effectively we only need protection for the first few seconds until the member
+            # key was updated with the new state/role values after demotion has started.
+            if task.timeout is None\
+                and (i is not None and self._tasks[i].timeout is not None
+                     or self._in_flight and self._in_flight.group == task.group
+                     and self._in_flight.deadline > time.time()):
```

Comment:
This condition is pretty long and thus hard to parse for most humans. For these kinds of cases I prefer to split them up. Either in separate if statements, that simply do the same early return. Or I group them in names, e.g. 
```python
waiting_for_timeout = i is not None and self._tasks[i].timeout is not None
```